src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/AvoidUninstantiatedInternalClasses.cs

```diff
@@ -101,6 +103,17 @@ public override void Initialize(AnalysisContext analysisContext)
                     {
                         instantiatedTypes.TryAdd(type.BaseType, null);
                     }
+
+                    // Consider class type declared in the CoClass attribute as instantiated
+                    if (type.TypeKind == TypeKind.Interface &&
+                        type.GetAttributes().FirstOrDefault(x => x.AttributeClass.Equals(coClassAttributeSymbol)) is AttributeData coClassAttribute &&
+                        coClassAttribute.ConstructorArguments.Length == 1 &&
```

Comment:
I have tried to add checks to restrict cases where we would add the related type. Do you think that's enough? Or maybe overkill?